File: Strivers A2Z DSA Sheet/1903-LargestOddNumberInString/1903-LargestOddNumberInString.java
// Last updated: 9/8/2026, 2:32:15 PM
class Solution {
    public String largestOddNumber(String num) {
        char[] c = num.toCharArray();
        int max = -1;
        for(int i=c.length-1; i>=0; i--){
            if((c[i] - '0')%2 == 1){
                max = i;
                break;
            }
        }
        return max == -1 ? "" : new String(c, 0, max+1);
    }
}

File: Strivers A2Z DSA Sheet/1903-LargestOddNumberInString/1903-LargestOddNumberInString.cpp
// Last updated: 9/8/2026, 2:32:22 PM
class Solution {
public:
    string largestOddNumber(string num) {
        int end = -1;
        int i;

        for(int i  = num.size()-1; i>=0; i--){
            if((num[i] - '0')%2 == 1){
                end = i;
                break; 
            }
        }
        

        i = 0;
        while(i<=end && num[i]=='0') i++;

        return num.substr(i, end - i+1);
    }
};